tests/main/snap-userd/task.yaml

```diff
@@ -0,0 +1,118 @@
+summary: Ensure snap userd allows opening a URL via xdg-open
+
+systems:
+    # Not supposed to work on Ubuntu Core systems as we don't have
+    # a user session environment there
+    - -ubuntu-core-*
+
+environment:
+    DISPLAY: :0
+
+restore: |
+    . "$TESTSLIB/pkgdb.sh"
+    rm -f dbus.env
+    if [[ "$SPREAD_SYSTEM" == ubuntu-14.04-* ]]; then
+        stop test-snap-userd || true
+        rm -f /etc/init/test-snap-userd.conf
+    else
+        systemctl stop --signal=KILL test-snap-userd.scope || true
+    fi
+    umount -f /usr/bin/xdg-open || true
+    umount -f /snap/core/current/usr/bin/xdg-open || true
+    distro_purge_package dbus-x11 xdg-utils
+
+execute: |
+    . "$TESTSLIB/pkgdb.sh"
+    . "$TESTSLIB/dirs.sh"
+
+    # Install necessary pacakges to get dbus-launch helper
+    distro_install_package dbus-x11 xdg-utils
+
+    dbus-launch > dbus.env
+    export $(cat dbus.env | xargs)
+
+    # helper that returns true when io.snapcraft.Launcher.OpenURL
+    # responds
+    ping_safe_launcher() {
+        dbus-send --session                                        \
+            --dest=io.snapcraft.Launcher                           \
+            --type=method_call                                     \
+            --print-reply                                          \
+            /io/snapcraft/Launcher                                 \
+            io.snapcraft.Launcher.OpenURL string:"ping-ping-ping:" \
```

Comment:
Please use the regular peer ping method here.

Reply:
Unfortuantely ping is not implemented by us, I can dig a bit more into this but it looks like a limiation/missing feature of the godbus library which is not very high-level.

Reply:
I changed the code to use "ping" now.